### DISCUSSION TITLE
Client config file

### DISCUSSION THREAD

**Discussion by janaakhterov** [hashgraph/hedera-sdk-java#574] [Ideas]
The current config for client looks something like
```json
{
    "operator": {
        "accountId": "<id>",
        "privateKey": "<key>"
    },
    "network": "testnet",
    "mirrorNetwork": "testnet"
}
```

However, `Client` has so many other configuration options that are not included here, and the list grows frequently.
 - `maxTransactionFee`
 - `maxQueryPayment`
 - `signOnDemand`
 - `requestTimeout`
 - `closeDuration`
 - `maxAttempts`
 - `nodeWaitTime`
 - `maxNodeAttempts`
 - `maxNodesPerTransaction`

What I propose is two things:
1) Support all the aforementioned fields in the config.
```json
{
    "operator": {
        "accountId": "<id>",
        "privateKey": "<key>"
    },
    "network": "testnet",
    "mirrorNetwork": "testnet",
    "maxTransactionFee": "2 ℏ",
    "maxQueryPayment": "2 ℏ"
}
```

2) Create a new type called `Config` which contains all these fields, and update `Client` to pass-through the setters and getters and/or implement `Client.getConfig()` which returns the `Config` allowing the user to easily modify it.

```java
public class Config {
    private Hbar maxTransactionFee;
    private Hbar maxQueryPayment;
    private Duration requestTimeout;
    private Duration closeDuration;
    private Duration nodeWaitTime;

    private boolean signOnDemand;

    private int maxAttempts;
    private int maxNodeAttempts;
    private int maxNodesPerTransaction;


    public Config fromString(String json) {
        // parse configuration file
    }

    public Config setMaxTransactionFee(Hbar maxTransactionFee) {
        this.maxTransactionFee = maxTransactionFee;
        return this;
    }

    public Hbar getMaxTransactionFee() {
        return this.maxTransactionFee;
    }

    // ...
}
```

**Comment by valtyr-naut** [comment]
It might be easier to just add/modify fields in the config. On the other hand it might be easier for developers to use/understand a type where comments can be added.

**Reply by valtyr-naut** [reply]
kind of a toss up

**Comment by Sean-Tedrow-LB** [comment]
Is it currently possible to specify a network name for a custom network in config file?  I think it currently is not.  That's something that needs to be added.
